### PR TITLE
Fix infinite loop with nested model and :attributes key

### LINKED ISSUE #78: Using key `attributes` with nested model causes infinite loop when attributes' value is hash with incorrect keys
When the `attributes` key is used with a hash, and the hash keys are unknown,  an infinite loop is triggered in the [unknown attributes handler](https://github.com/DmitryTsepelev/store_model/blob/d5b766916f01007fbf295588293ad702a0caf7a2/lib/store_model/types/one_base.rb#L60). `value_symbolized.except(:attributes)` doesn't remove `:attributes`, and instead tries to remove the incorrect key _inside_ the attributes hash.

```ruby
class Configuration
  include StoreModel::Model

  attribute :color, :string
end

class ConfigurationWithNesting
  include StoreModel::Model
  attribute :configuration, Configuration.to_type
end


# Non-nested creation with incorrect attribute, with attributes: key
Configuration.new(attributes: { test: 123 })
# Errors correctly
# ActiveModel::UnknownAttributeError: unknown attribute 'test' for Configuration.

# Creation with correct attribute, with attributes: key
ConfigurationWithNesting.new(configuration: { attributes: { color: 123 } })
# Works correctly
# #<ConfigurationWithNesting configuration: #<Configuration:0x00005588e7084c40>>

# Creation with incorrect attribute, with attributes: key
ConfigurationWithNesting.new(configuration: { attributes: { test: 123 } })
# Infinite loop
```

After interrupting:

```
(pry) output error: Interrupt
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:52:in `_assign_attribute'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:43:in `block in _assign_attributes'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:42:in `each'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:42:in `_assign_attributes'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:35:in `assign_attributes'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:50:in `public_send'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:50:in `_assign_attribute'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:43:in `block in _assign_attributes'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:42:in `each'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:42:in `_assign_attributes'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attribute_assignment.rb:35:in `assign_attributes'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/model.rb:81:in `initialize'
/usr/local/bundle/gems/activemodel-6.0.3.4/lib/active_model/attributes.rb:79:in `initialize'
/usr/local/bundle/gems/store_model-0.8.0/lib/store_model/types/one.rb:65:in `new'
/usr/local/bundle/gems/store_model-0.8.0/lib/store_model/types/one.rb:65:in `model_instance'
/usr/local/bundle/gems/store_model-0.8.0/lib/store_model/types/one.rb:33:in `cast_value'
/usr/local/bundle/gems/store_model-0.8.0/lib/store_model/types/one_base.rb:60:in `handle_unknown_attribute'
/usr/local/bundle/gems/store_model-0.8.0/lib/store_model/types/one.rb:38:in `rescue in cast_value'

... loops endlessly after this
```

In both cases, `value_symbolized.except(attribute)` tries to remove `:test`, but in the nested case, `:attributes` should be removed instead. This seems to be because `ActiveModel::UnknownAttributeError` is not raised with `:attributes`.

### PR DESCRIPTION
**Pull request by timhwang21**
Fixes #78.

This PR shows a proof-of-concept fix, and I'm happy to iterate on this to get a solution you like. I'm not familar with the codebase and am not sure if this is the most comprehensive fix.

### CODE REVIEW

**Comment by DmitryTsepelev** [comment]
Hi @timhwang21, thank you for the PR! Could you please explain why there is a need to pass `attributes` as a has key (e.g., it could be reduced to`Configuration.new(test: 123)`)? Looks like I'm missing something

**Comment by timhwang21** [comment]
@DmitryTsepelev there's no "good reason" to, but the API supports it. At least for me, it happened accidentally (I had an attribute `config` which I misremembered as `attributes`, thus causing a crash when editing). I personally think getting rid of this Active Model attributes hash behavior would be safer and clearer, but I think it would be a breaking change!

File: lib/store_model/types/one_base.rb [hunk 1]
```diff
@@ -57,8 +57,18 @@ def handle_unknown_attribute(value, exception)
         attribute = exception.attribute.to_sym
         value_symbolized = value.symbolize_keys
 
-        cast_value(value_symbolized.except(attribute)).tap do |configuration|
+        filtered_value = value_symbolized.except(attribute)
+
+        if value_symbolized.key?(:attributes)
+          filtered_value[:attributes] = filtered_value[:attributes].except(attribute)
```
Comment (DmitryTsepelev) [inline lib/store_model/types/one_base.rb:63]: I guess we could make this check before `.exept`, i.e.

```ruby
value_symbolized = value_symbolized[:attributes] if value_symbolized.key?(:attributes)
filtered_value = value_symbolized.except(attribute)
```

**Review by DmitryTsepelev — CHANGES_REQUESTED** [review]
Makes sense, thanks for catching that! I've left a small comment, could you please try if we could simplify the code a bit?

Also, could you please check rubocop (it will probably go away after the change I mentioned above) and add yourself to the changelog before I merge it in?

File: lib/store_model/types/one_base.rb (hunk 1 above)
Comment (timhwang21) [reply lib/store_model/types/one_base.rb:63]: Cool, that makes sense to me (though I think we should be clear that mixing `attributes` and other top-level values isn't possible... will add a spec)

**Comment by DmitryTsepelev** [comment]
LGTM, thanks!

**Review by DmitryTsepelev — APPROVED** [review]

**Pull request merged**